tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -1962,6 +3051,109 @@ void test_lte_lc_periodic_search_get_success(void)
 	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
 }
 
+void test_lte_lc_periodic_search_get_success_patterns4(void)
+{
+	int ret;
+	struct lte_lc_periodic_search_cfg cfg;
+
+	__mock_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT%PERIODICSEARCHCONF=1",
+		"%%PERIODICSEARCHCONF: %hu,%hu,%hu,\"%40[^\"]\",\"%40[^\"]\",\"%40[^\"]\",\"%40[^\"]\"",
+		7);
+	__mock_nrf_modem_at_scanf_ReturnVarg_uint16(0); /* loop */
+	__mock_nrf_modem_at_scanf_ReturnVarg_uint16(0); /* return_to_pattern */
+	__mock_nrf_modem_at_scanf_ReturnVarg_uint16(10); /* band_optimization */
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("0,60,3600,,600"); /* pattern_buf */
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("0,30,1800,20,48"); /* pattern_buf */
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("1,10,20,30,40,50"); /* pattern_buf */
+	__mock_nrf_modem_at_scanf_ReturnVarg_string("1,11,21,31"); /* pattern_buf */
+
+	ret = lte_lc_periodic_search_get(&cfg);
+	TEST_ASSERT_EQUAL(0, cfg.loop);
+	TEST_ASSERT_EQUAL(0, cfg.return_to_pattern);
+	TEST_ASSERT_EQUAL(10, cfg.band_optimization);
+	TEST_ASSERT_EQUAL(LTE_LC_PERIODIC_SEARCH_PATTERN_RANGE, cfg.patterns[0].type);
+	TEST_ASSERT_EQUAL(60, cfg.patterns[0].range.initial_sleep);
+	TEST_ASSERT_EQUAL(3600, cfg.patterns[0].range.final_sleep);
+	TEST_ASSERT_EQUAL(-1, cfg.patterns[0].range.time_to_final_sleep);
+	TEST_ASSERT_EQUAL(600, cfg.patterns[0].range.pattern_end_point);
+	TEST_ASSERT_EQUAL(LTE_LC_PERIODIC_SEARCH_PATTERN_RANGE, cfg.patterns[1].type);
+	TEST_ASSERT_EQUAL(30, cfg.patterns[1].range.initial_sleep);
+	TEST_ASSERT_EQUAL(1800, cfg.patterns[1].range.final_sleep);
+	TEST_ASSERT_EQUAL(20, cfg.patterns[1].range.time_to_final_sleep);
+	TEST_ASSERT_EQUAL(48, cfg.patterns[1].range.pattern_end_point);
+	TEST_ASSERT_EQUAL(LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE, cfg.patterns[2].type);
+	TEST_ASSERT_EQUAL(10, cfg.patterns[2].table.val_1);
+	TEST_ASSERT_EQUAL(20, cfg.patterns[2].table.val_2);
+	TEST_ASSERT_EQUAL(30, cfg.patterns[2].table.val_3);
+	TEST_ASSERT_EQUAL(40, cfg.patterns[2].table.val_4);
+	TEST_ASSERT_EQUAL(50, cfg.patterns[2].table.val_5);
+	TEST_ASSERT_EQUAL(LTE_LC_PERIODIC_SEARCH_PATTERN_TABLE, cfg.patterns[3].type);
+	TEST_ASSERT_EQUAL(11, cfg.patterns[3].table.val_1);
+	TEST_ASSERT_EQUAL(21, cfg.patterns[3].table.val_2);
+	TEST_ASSERT_EQUAL(31, cfg.patterns[3].table.val_3);
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+}
+
+void test_lte_lc_reduced_mobility_set_success(void)
+{
+	int ret;
+
+	__mock_nrf_modem_at_printf_ExpectAndReturn("AT%REDMOB=0", 0);
```

Comment:
Rather than hardcoding the integer value to be sent, perhaps it would be better to sprintf the string using the actual mode value. This way if they get changed or reordered, the tests don't need to be updated.
Same for similar cases.

Reply:
I specifically want to hardcode the integer value. This makes the test more readable in my opinion showing clearly the AT command that we are expecting lte_lc to send. We also avoid putting the same sprintf code into test case and real code.
More importantly, this integer is something that is specified in AT specs so we must not change it. Having said that, if a change would still be needed, we can modify the test case.
So, I'm trying to say that I want this test case to fail if somebody modifies the values of `lte_lc_reduced_mobility_mode` and same applies for all other AT commands and integers in there. I've used sprintf few times in some loops and Kconfig values.

Reply:
That makes sense, thank you.